### PR TITLE
Add ipVersion field to support IPv6 in global address and release to Terraform

### PR DESCRIPTION
**Pull request by rosbo**
- Add new ipVersion field.
- Add documentation links to GlobalAddress
- Tweak property documentation to remove the mention of some fields being only allowed in global address. It is redundant. This resource is GlobalAddress. There is a different resource for Address
- Terraform
  - Support Enum validation
  - Release generated GlobalAddress

-----------------------------------------------------------------
# [all]
## [terraform]
Release generated GlobalAddress
### [puppet-compute]
Add ipVersion property to support IPv6

### CODE REVIEW

File: products/compute/terraform.yaml [hunk 1]
```diff
@@ -344,6 +349,9 @@ files: !ruby/object:Provider::Config::Files
     'google/transport_test.go': 'templates/terraform/transport_test.go'
     'google/import.go': 'templates/terraform/import.go'
     'google/import_test.go': 'templates/terraform/import_test.go'
+    # Handwritten acceptance tests for autogenerated resources.
+    # Adding them here allows updating the tests as part of a MM pull request.
+    'google/resource_compute_global_address_test.go': 'templates/terraform/tests/resource_compute_global_address_test.go'
```
Comment (nelsonjr) [inline products/compute/terraform.yaml]: Try to keep the YAML files 80-chars. It works well if you break this line

File: products/compute/api.yaml [hunk 2]
```diff
@@ -899,6 +899,10 @@ objects:
     description: |
       Represents a Global Address resource. Global addresses are used for
       HTTP(S) load balancing.
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Reserving a Static External IP Address': 'https://cloud.google.com/compute/docs/ip-addresses/reserve-static-external-ip-address'
```
Comment (nelsonjr) [inline products/compute/api.yaml]: Break line at ":"

File: products/compute/terraform.yaml (hunk 1 above)
Comment (rosbo) [reply products/compute/terraform.yaml]: Done

File: products/compute/api.yaml (hunk 2 above)
Comment (rosbo) [reply products/compute/api.yaml]: Done

**Pull request merged**